appserver/ejb/ejb-http-remoting/endpoint/pom.xml

```diff
@@ -86,6 +86,36 @@
             <version>${project.version}</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>commons-io</groupId>
```

Comment:
Anyway this can be done without adding the commons dependency?

Reply:
I use if for the `ClassLoaderObjectInputStream`. Its Apache licensed so I guess we should be allowed to only grab that class?

We already do use `commons-io` elsewhere so if we don't want to use it we should probably remove it everywhere? Maybe you can help me understand why this dependency is unwanted while we have lots of upstream dependencies. Is it because its a general utility that might cause trouble in case applications use it in a different version?

Reply:
If we are using it anyway then I'm fine with it.

Reply:
In general there are a number of reasons we don't want to bring in new 3rd party dependencies lightly;

1. It bloats the server
2. Applications may use a different version and therefore it can cause problems
3. It could increase the attack surface of the server. 
4. They need a license review and update to all the 3rd party license files

Reply:
As discussed on Teams we avoid adding `commons.io` by repackaging the needed class.

Reply:
Backed adding as repackaged class since removing `commons.io` from nucleus results in errors. It is needed. Created PAYARA-3923 to investigate where it is used and if we could remove it.